README.md

```diff
@@ -1 +1,649 @@
+| `type` | `"awake/msg"`                     | Generic AWAKE message type                                     | Yes      |
+| `id`   | `sha3_256(resEcdhPk + reqEcdhPk)` | Message ID                                                     | Yes      |
+| `iv`   |                                   | Initialization vector for the encrypted payload                | Yes      |
+| `msg`  |                                   | Fulfilled challenge payload encrypted with Step 4 ECDH AES-key | Yes      |
+
+#### 3.5.1.1 Encrypted Message
+
+The encrypted message payload MUST include an `awake/ack` field, with a value of the requestor's long-term DID. This payload MAY contain additional fields. This is often useful if dovetailing the ACK with the first message of a session using the 
+
+``` javascript
+{
+  ...,
+  "awake/ack": reqActualDid
+}
+```
+
+## 4 Secure Session
+
+```
+Requestor                  Responder
+    ⋮                          ⋮
+    │                          │
+    │         Messages         │ (6)
+    │◄────────────────────────►│
+    │                          │
+    ⋮                          ⋮
+```
+
+Messages sent over an established AWAKE session MUST contain the following keys:
+ 
+| Field  | Value                                                 | Description                                                             | Required |
+| ------ | ----------------------------------------------------- | ----------------------------------------------------------------------- | -------- |
+| `awv`  | `"0.1.0"`                                             | AWAKE message version                                                   | Yes      |
+| `type` | `"awake/msg"`                                         | Generic AWAKE message type                                              | Yes      |
+| `mid`  | `sha3_256(latestSenderEcdhPk + latestReceiverEcdhPk)` | Message ID                                                              | Yes      |
+| `sid`  | `sha3_256(firstAesKey)`                               | OPTIONAL session ID                                                     | No       |
+| `iv`   |                                                       | Initialization vector for the encrypted payload                         | Yes      |
+| `msg`  |                                                       | Fulfilled challenge payload encrypted with latest KDF AES-key           | Yes      |
+| `esig` | `encrypt(messageKey, sig(senderUcanSk, msg + mid))`   | OPTIONAL message signature using the sender's UCAN-validated public key | No       |
+
+The OPTIONAL encrypted signature field (`esig`) MAY be included to prove authenticity of the payload. Without this field, an eavesdropper with access to the input secret MAY be able to spoof a message that updates the next ECDH key to one it controls. The signature MUST be encrypted, since some commonly used key types such as RSA allow for the derivation of the signer's public key.
```

Comment:
I'm having a little bit of trouble visualizing this attack. Do you have a situation in mind where an eavesdropper will have access to the input secret, but without having compromised the device in a way that allows them to produce this signature themselves?

I ask because the `esig` field being optional raises some alarms for me: this is the sort of thing implementers would skip in a first version, so if it's an important mitigation, maybe it shouldn't be optional. And if it actually isn't required, it might be helpful to include a little bit more guidance on making that decision.

Reply:
haha I had a feel that this was going to come up 😅 

The scenario is that if an attacker manages to spoof a single message, they can _change_ the next ECDH key. Signing with the UCAN DID key avoids this because it authenticates the message without depending on the AES envelope. Signing can't hurt, but also break-in security is the point of the double ratchet.

That said, when writing this section, it did feel like a bridge too far / like I was wearing a stack of tinfoil hats.

You're 100% right that hedging with a SHOULD is the wrong strategy here.

I'll go ahead and drop this field. Thanks for the rubber ducking 🦆